WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/BadgeView.swift

```diff
@@ -1,42 +1,78 @@
 import SwiftUI
 
 struct BadgeView: View {
-    enum BadgeType {
+    enum BadgeType: Equatable {
         case new
         case tip
+        case remoteImage(lightUrl: URL, darkUrl: URL?)
+        case customText(text: String)
 
-        var title: String {
+        var title: String? {
             switch self {
             case .new:
                 return Localization.newTitle
             case .tip:
                 return Localization.tipTitle
+            case .customText(let text):
+                return text
+            case .remoteImage:
+                return nil
+            }
+        }
+
+        static func ==(lhs: BadgeType, rhs: BadgeType) -> Bool {
+            switch (lhs, rhs) {
+            case (.new, .new), (.tip, .tip):
+                return true
+            case (.customText(let lhsText), .customText(let rhsText)):
+                return lhsText == rhsText
+            case (remoteImage(let lhsLightUrl, let lhsDarkUrl), remoteImage(let rhsLightUrl, let rhsDarkUrl)):
+                return lhsLightUrl == rhsLightUrl && lhsDarkUrl == rhsDarkUrl
+            default:
+                return false
```

Comment:
Is this function removable? So far we do not seem to need the `==` operator for `BadgeType` equatable conformance, as all its properties are already equatable, and we can compare them without it.

Reply:
I was also surprised that I couldn't build until I added this, and did clean/restart because it seemed wrong before writing it out. Now it seems like the compiler just didn't feel like synthesising it for me last week. A new week, a new start, it's being much more co-operative!

Removed in 25941e1